src/test/java/tests/common_test/BaseTest.java

```diff
@@ -0,0 +1,36 @@
+package tests.common_test;
+
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.*;
+import utilities.DriverManagerFactory;
+import utilities.DriverManager;
+
+public class BaseTest {
+
+    private DriverManager driverManager;
+    private WebDriver webDriver;
+
+    @Parameters({"browser"})
+
+    @BeforeClass
+    public void beforeTest(@Optional("chrome") String browser) {
+        driverManager = DriverManagerFactory.getManager(browser);
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        webDriver = driverManager.getInstance();
+    }
+
+    @AfterMethod
+    public void closeWebDriver() {
+        if (webDriver != null) {
+            webDriver.quit();
+        }
+        webDriver = null;
+    }
```

Comment:
We don't need to set webDriver = null after executing webDriver.close() - java has Garbage Collector, which clean up all unnecessary instances.

Reply:
Fixed

Reply:
Approved